File: app/src/main/java/com/jurtz/marcel/shoppinglist/database/ShoppingListDao.java
package com.jurtz.marcel.shoppinglist.database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.jurtz.marcel.shoppinglist.model.ShoppingList;

import java.util.List;

@Dao
public interface ShoppingListDao {

    @Query("SELECT * FROM shopping_list")
    List<ShoppingList> getAll();

    @Query("DELETE FROM shopping_list WHERE id = :id")
    void deleteListById(int id);

    @Insert()
    void insertList(ShoppingList shoppingList);

    @Update()
    void updateList(ShoppingList shoppingList);

    @Delete
    void deleteList(ShoppingList shoppingList);
}

File: app/src/main/java/com/jurtz/marcel/shoppinglist/database/ShoppingListDAO.kt
package com.jurtz.marcel.shoppinglist.database

import android.arch.persistence.room.*
import com.jurtz.marcel.shoppinglist.model.ShoppingList

@Dao
interface ShoppingListDAO {
    @Query("SELECT * FROM shopping_list")
    fun getAll() : List<ShoppingList>

    @Query("SELECT * FROM shopping_list WHERE id = :p0")
    fun findTaskById(id: Long): ShoppingList

    @Insert()
    fun insertTask(shoppingList: ShoppingList)

    @Update()
    fun updateTask(shoppingList: ShoppingList)

    @Delete
    fun deleteTask(shoppingList: ShoppingList)
}